pkg/agent/openflow/pipeline.go

```diff
@@ -2704,3 +2655,233 @@ func (sl conjunctiveActionsInOrder) Less(i, j int) bool {
+				MatchDstIP(nodeIP).
+				MatchRegMark(AntreaFlexibleIPAMRegMark).
+				Action().SetDstMAC(f.nodeConfig.UplinkNetConfig.MAC).
+				Action().GotoStage(stageSwitching).
+				Done(),
+			// When Node bridge local port and uplink port connect to OVS, this generates the flow to match the reply
+			// packets of connection initiated through the bridge local port with FromBridgeCTMark.
+			L3ForwardingTable.ofTable.BuildFlow(priorityHigh).
+				Cookie(cookieID).
+				MatchProtocol(ipProtocol).
+				MatchCTMark(FromBridgeCTMark).
+				MatchCTStateRpl(true).
+				MatchCTStateTrk(true).
+				Action().SetDstMAC(f.nodeConfig.UplinkNetConfig.MAC).
+				Action().GotoStage(stageSwitching).
+				Done())
+	}
+	return flows
+}
+
+// l3FwdFlowToExternal generates the flow to match the packets of non-Service destined for external network.
+func (f *featurePodConnectivity) l3FwdFlowToExternal() binding.Flow {
+	// TODO: load ToUplinkRegMark for packets from Antrea IPAM Pod.
+	return L3ForwardingTable.ofTable.BuildFlow(priorityMiss).
+		Cookie(f.cookieAllocator.Request(f.category).Raw()).
+		Action().LoadRegMark(ToGatewayRegMark).
+		Action().GotoStage(stageSwitching).
+		Done()
+
+}
+
+// hostBridgeLocalFlows generates the flows to match the packets forwarded between bridge local port and uplink port.
+func (f *featurePodConnectivity) hostBridgeLocalFlows() []binding.Flow {
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	return []binding.Flow{
+		// This generates the flow to forward the packets from uplink port to bridge local port.
+		ClassifierTable.ofTable.BuildFlow(priorityNormal).
+			Cookie(cookieID).
+			MatchInPort(config.UplinkOFPort).
+			Action().Output(config.BridgeOFPort).
+			Done(),
+		// This generates the flow to forward the packets from bridge local port to uplink port.
+		ClassifierTable.ofTable.BuildFlow(priorityNormal).
+			Cookie(cookieID).
+			MatchInPort(config.BridgeOFPort).
+			Action().Output(config.UplinkOFPort).
+			Done(),
+	}
+}
+
+// preRoutingClassifierFlows generates the flow to classify packets in stagePreRouting.
+func (f *featureService) preRoutingClassifierFlows() []binding.Flow {
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	var flows []binding.Flow
+
+	targetTables := []uint8{SessionAffinityTable.ofTable.GetID(), ServiceLBTable.ofTable.GetID()}
+	if f.proxyAll {
+		targetTables = append([]uint8{NodePortMarkTable.ofTable.GetID()}, targetTables...)
+	}
+	for _, ipProtocol := range f.ipProtocols {
+		flows = append(flows,
+			// This generates the default flow to match the first packet of Service connection.
+			PreRoutingClassifierTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
+				MatchProtocol(ipProtocol).
+				Action().ResubmitToTables(targetTables...).
+				Done(),
+		)
+	}
+
+	return flows
+}
+
+// l3FwdFlowsToExternal generates the flows to forward the packets of Service connection to external network.
+func (f *featureService) l3FwdFlowsToExternal() []binding.Flow {
```

Comment:
The function name doesn't reflect what it does clearly. For example, hard to imagine it handles Service connection only from the name. Maybe l3FwdFlowsToExternalEndpoint is more proper

Reply:
Agreed.